### PR TITLE
Update object image code to use new DB structure

### PR DESCRIPTION
**Pull request by brickbots**
Also include fix to logging system crash I found while testing.

### CODE REVIEW

File: python/PiFinder/setup.py [hunk 1]
```diff
@@ -163,8 +164,67 @@ def insert_catalog_max_sequence(catalog_name):
     conn.commit()
 
 
+def resolve_object_images():
+    # This is the list of catalogs to search for
+    # objects to match against image names
+    resolution_priority = [
```
Comment (mrosseel) [inline python/PiFinder/setup.py:170]: nothing wrong with this for now, but would like to have an extra column in the catalog table with the priority for that catalog

File: python/PiFinder/setup.py [hunk 2]
```diff
@@ -163,8 +164,67 @@ def insert_catalog_max_sequence(catalog_name):
...
+            if catalog_check:
+                # Found a match!
+                resolved_name = f"{catalog_code}{catalog_check['sequence']}"
+                break
+
+        if resolved_name == None:
+            # Didn't find a name in the priority list
+            # Pick one from the remainders, but sort
+            # by catalog code so this is deterministic
+            catalog_entry = db_c.execute(
```
Comment (mrosseel) [inline python/PiFinder/setup.py:205]: why not objects_db.get_catalog_object_by_object_id() ?

**Review by mrosseel — CHANGES_REQUESTED** [review]

File: python/PiFinder/setup.py [hunk 3]
```diff
@@ -163,8 +164,67 @@ def insert_catalog_max_sequence(catalog_name):
...
+    # objects to match against image names
+    resolution_priority = [
+        "NGC",
+        "IC",
+        "M",
+        "C",
+    ]
+    # load all objects in objects table
+    conn, db_c = objects_db.get_conn_cursor()
+    all_objects = db_c.execute(
```
[the remark this answers is not in the document]
Comment (brickbots) [reply python/PiFinder/setup.py]: Indeed!  I've converted this over

File: python/PiFinder/setup.py (hunk 1 above)
Comment (brickbots) [reply python/PiFinder/setup.py]: I've changed this to use the insert order of the catalogs for priority.  I slightly adjusted the catalog import order and added a comment to remind future folks that the order is important

File: python/PiFinder/setup.py (hunk 2 above)
Comment (brickbots) [reply python/PiFinder/setup.py]: Rewrote this to just continue down the priority order until it finds one which currently yields the same output.

**Comment by brickbots** [comment]
Thanks for the review @mrosseel, some great suggestions that caused me to rethink a bit and turn out better code 👍

**Pull request merged**